crates/viewer/re_ui/src/color_table.rs

```diff
@@ -0,0 +1,217 @@
+    Purple,
+}
+
+impl nohash_hasher::IsEnabled for Hue {}
+
+impl Hue {
+    pub fn all() -> &'static [Self] {
+        static ALL: [Hue; 5] = [Hue::Gray, Hue::Green, Hue::Red, Hue::Blue, Hue::Purple];
+        &ALL
+    }
+}
+
+impl Display for Hue {
+    fn fmt(&self, f: &mut Formatter<'_>) -> std::fmt::Result {
+        match self {
+            // these must be as they appear in `design_token.json`
+            Self::Gray => f.write_str("Gray"),
+            Self::Green => f.write_str("Green"),
+            Self::Red => f.write_str("Red"),
+            Self::Blue => f.write_str("Blue"),
+            Self::Purple => f.write_str("Purple"),
+        }
+    }
+}
+
+/// A shade for a [`ColorToken`].
+#[derive(Debug, Clone, Copy, Hash, PartialEq, Eq)]
+pub enum Shade {
+    S0 = 0,
+    S25 = 25,
+    S50 = 50,
+    S75 = 75,
+    S100 = 100,
+    S125 = 125,
+    S150 = 150,
+    S175 = 175,
+    S200 = 200,
+    S225 = 225,
+    S250 = 250,
+    S275 = 275,
+    S300 = 300,
+    S325 = 325,
+    S350 = 350,
+    S375 = 375,
+    S400 = 400,
+    S425 = 425,
+    S450 = 450,
+    S475 = 475,
+    S500 = 500,
+    S525 = 525,
+    S550 = 550,
+    S575 = 575,
+    S600 = 600,
+    S625 = 625,
+    S650 = 650,
+    S675 = 675,
+    S700 = 700,
+    S725 = 725,
+    S750 = 750,
+    S775 = 775,
+    S800 = 800,
+    S825 = 825,
+    S850 = 850,
+    S875 = 875,
+    S900 = 900,
+    S925 = 925,
+    S950 = 950,
+    S975 = 975,
+    S1000 = 1000,
+}
+
+impl nohash_hasher::IsEnabled for Shade {}
+
+impl Shade {
+    pub fn all() -> &'static [Self] {
+        static ALL: [Shade; 41] = [
+            Shade::S0,
+            Shade::S25,
+            Shade::S50,
+            Shade::S75,
+            Shade::S100,
+            Shade::S125,
+            Shade::S150,
+            Shade::S175,
+            Shade::S200,
+            Shade::S225,
+            Shade::S250,
+            Shade::S275,
+            Shade::S300,
+            Shade::S325,
+            Shade::S350,
+            Shade::S375,
+            Shade::S400,
+            Shade::S425,
+            Shade::S450,
+            Shade::S475,
+            Shade::S500,
+            Shade::S525,
+            Shade::S550,
+            Shade::S575,
+            Shade::S600,
+            Shade::S625,
+            Shade::S650,
+            Shade::S675,
+            Shade::S700,
+            Shade::S725,
+            Shade::S750,
+            Shade::S775,
+            Shade::S800,
+            Shade::S825,
+            Shade::S850,
+            Shade::S875,
+            Shade::S900,
+            Shade::S925,
+            Shade::S950,
+            Shade::S975,
+            Shade::S1000,
+        ];
+        &ALL
+    }
+
+    pub fn as_u16(self) -> u16 {
+        self as u16
+    }
+}
+
+/// A table mapping all combination of [`Hue`] and [`Shade`] to a [`egui::Color32`].
+#[derive(Debug)]
+pub struct ColorTable {
+    pub color_table: nohash_hasher::IntMap<Hue, nohash_hasher::IntMap<Shade, egui::Color32>>,
```

Comment:
You're using `nohash_hasher` here, presumably for speed.

If you really want full speed, use sequential numbers of the enum variants (that's the default) and then use a LUT

Reply:
Do you have a suggestion for a pattern/helper crate for obtaining at compile time the number of variant of an enum and a list of all variants?

Reply:
https://doc.rust-lang.org/stable/std/mem/fn.variant_count.html :]

Reply:
that looks really cool! but...

<img width="414" alt="image" src="https://github.com/user-attachments/assets/227a1795-f30f-4b38-ac26-5401308cf0bc">

Reply:
oh I can use `strum`